### DISCUSSION TITLE
`livewire:form` ignores published stubs

### DISCUSSION THREAD

**Discussion by kokoshneta** [livewire/livewire#7855] [7. Bugs (archived)]
### Livewire version

v3.4.4

### Laravel version

v10.41.0

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

1. Publish Livewire stubs via `php artisan livewire:stubs`
2. Modify the published `livewire.form.stub` in some way
3. Create a form object with `php artisan livewire:form FormName`

Result: the created form class is based on the default stub, rather than the modified, published stub.

[The docs do say](https://livewire.laravel.com/docs/components#customizing-component-stubs), “you can still use the `make:livewire` Artisan command and Livewire will automatically use your custom stubs when generating files”, and `make:livewire` does indeed work.

But the Artisan command for making a form is `livewire:form` rather than `make:livewire`, and although it’s not listed on the stubs page in the docs, publishing stubs does create a `livewire.form.stub` file as well, so presumably the intention is that form object creation should take published stubs into account just like regular component creation does.

(No repo/Wirebox, since no user code is required in a clean install prior to steps 1–3 above.)

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@kokoshneta ~~this has been fixed in PR #7816 and will be in the next release 🙂~~

Edit: Sorry, the publishing has been released in v3.4.4. Hmm I wonder why it's not using them correctly 🤔

**Reply by kokoshneta** [reply]
I would guess probably because the check to see if there are published stubs before falling back on the default ones only happens in the `make:livewire` command? I haven’t delved into the actual commands, but the fact that forms don’t use `make:livewire` makes me suspect that that’s the reason.

**Reply by kokoshneta** [reply]
Yup, that looks like what it is. Looks like [`FormCommand::getStub()`](https://github.com/livewire/livewire/blob/4bf50fe80d581bcc95f6f96ea046cbeb1d2a076e/src/Features/SupportConsoleCommands/Commands/FormCommand.php#L35) has a hard-coded `__DIR__`.

(Not sure how exactly the stub path is calculated in the `MakeCommand` class, but it doesn’t look like it’s the same way.)

**Reply by joshhanley** [reply]
@kokoshneta yeah I think you are right. The make command uses the `ComponentParser` under the hood, which has the stubs directory check

https://github.com/livewire/livewire/blob/dc8ad6e08fa207944f26b8821776c156e88b460d/src/Features/SupportConsoleCommands/Commands/ComponentParser.php#L92-L98

So something similar would need to be added to the `livewire:form` command. PRs to fix are welcome! 🙂
